spec/datetimeformat.html

```diff
@@ -123,15 +123,33 @@ <h1>InitializeDateTimeFormat ( _dateTimeFormat_, _locales_, _options_ )</h1>
             1. Throw a *RangeError* exception.
           1. Let _timeZone_ be CanonicalizeTimeZoneName(_timeZone_).
         1. Set _dateTimeFormat_.[[TimeZone]] to _timeZone_.
-        1. Let _opt_ be a new Record.
+        1. Let _formatOptions_ be a new Record.
         1. For each row of <emu-xref href="#table-datetimeformat-components"></emu-xref>, except the header row, in table order, do
           1. Let _prop_ be the name given in the Property column of the row.
           1. If _prop_ is *"fractionalSecondDigits"*, then
             1. Let _value_ be ? GetNumberOption(options, *"fractionalSecondDigits"*, 1, 3, *undefined*).
           1. Else,
             1. Let _value_ be ? GetOption(_options_, _prop_, *"string"*, &laquo; the strings given in the Values column of the row &raquo;, *undefined*).
-          1. Set _opt_.[[&lt;_prop_&gt;]] to _value_.
+          1. Set _formatOptions_.[[&lt;_prop_&gt;]] to _value_.
         1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
+        1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
+        1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
+        1. If _hc_ is *null*, then
+          1. Set _hc_ to _hcDefault_.
+        1. If _hour12_ is *true*, then
+          1. If _hcDefault_ is *"h11"* or *"h23"*, then
+            1. Set _hc_ to *"h11"*.
+          1. Else,
+            1. Set _hc_ to *"h12"*.
+        1. Else if _hour12_ is *false*, then
+          1. If _hcDefault_ is *"h11"* or *"h23"*, then
+            1. Set _hc_ to *"h23"*.
+          1. Else,
+            1. Set _hc_ to *"h24"*.
+        1. Else,
+          1. Assert: _hour12_ is *undefined*.
+        1. Set _dateTimeFormat_.[[HourCycle]] to _hc_.
```

Comment:
Also, ugh, this is kind of a mess. Since this step _always_ overrides the value of _dateTimeFormat_.[[HourCycle]] that was set from _r_.[[hc]] above, I think that earlier step is misleading and should be replaced with `Let _hc_ be _r_.[[hc]].`. There's probably also a better overall ordering, but I think this is probably good enough for now.

Reply:
I can't make the suggestion contextual, but it changes line 114 and replaces (new) line 136.
```patch
         1. Set _dateTimeFormat_.[[Locale]] to _r_.[[locale]].
         1. Let _calendar_ be _r_.[[ca]].
         1. Set _dateTimeFormat_.[[Calendar]] to _calendar_.
-        1. Set _dateTimeFormat_.[[HourCycle]] to _r_.[[hc]].
+        1. Let _hc_ be _r_.[[hc]].
         1. Set _dateTimeFormat_.[[NumberingSystem]] to _r_.[[nu]].
         1. Let _dataLocale_ be _r_.[[dataLocale]].
         1. Let _timeZone_ be ? Get(_options_, *"timeZone"*).
```
```patch
           1. Set _formatOptions_.[[&lt;_prop_&gt;]] to _value_.
         1. Let _dataLocaleData_ be _localeData_.[[&lt;_dataLocale_&gt;]].
         1. Let _hcDefault_ be _dataLocaleData_.[[hourCycle]].
-        1. Let _hc_ be _dateTimeFormat_.[[HourCycle]].
         1. If _hc_ is *null*, then
           1. Set _hc_ to _hcDefault_.
         1. If _hour12_ is *true*, then
```